app/models/targeting.rb

```diff
@@ -0,0 +1,41 @@
+class Targeting < ActiveRecord::Base
+
+  STATIC_TYPE = 'static_query'
+  DYNAMIC_TYPE = 'dynamic_bookmark'
+  TYPES = {STATIC_TYPE => N_('Static Query'), DYNAMIC_TYPE => N_('Dynamic Bookmark')}
+
+  belongs_to :user
+  belongs_to :bookmark
+
+  has_many :targeting_hosts, :dependent => :destroy
+  has_many :hosts, :through => :targeting_hosts
+  has_many :template_invocations, :dependent => :destroy
+
+  validates :targeting_type, :presence => true, :inclusion => Targeting::TYPES.keys
+  validates :bookmark, :presence => true, :if => "dynamic?"
+
+  before_create :assign_search_query, :if => :static?
```

Comment:
Yeah, i think we should add it later and really once host collections (or whatever it will be called), is in core that may be a better solution for a static list?